### PR TITLE
Fix capsule collision bug

### PR DESCRIPTION
**Pull request by biotinker**

### CODE REVIEW

**Comment by biotinker** [comment]
@nfranczak

File: spatialmath/capsule.go [hunk 1]
```diff
@@ -112,14 +112,17 @@ func (c *capsule) AlmostEqual(g Geometry) bool {
 // Transform premultiplies the capsule pose with a transform, allowing the capsule to be moved in space.
 func (c *capsule) Transform(toPremultiply Pose) Geometry {
 	newPose := Compose(toPremultiply, c.pose)
+	segB := Compose(toPremultiply, NewPoseFromPoint(c.segB)).Point()
+	center := newPose.Point()
```
Comment (raybjork) [inline spatialmath/capsule.go:116]: actually not sure why we are storing this at all given that we have a `Pose`.  The caching speed up has got to be super minor right?

**Review by raybjork — APPROVED** [review]
LGTM.  Could you provide some context for how you found this?

File: spatialmath/capsule.go (hunk 1 above)
Comment (biotinker) [reply spatialmath/capsule.go:116]: Cacheing speedup was minor but definitely noticeable, 4% or so. The collision computations are *the* bottleneck for motion planning, so this is one of the few places it's worth it (others being FK calculations and nlopt)

**Review by spieswl — APPROVED** [review]
Looks good!

**Pull request merged**